tests/test_qubit_device.py

```diff
@@ -498,10 +498,22 @@ def test_non_analytic_expval(self, mock_qubit_device_with_original_statistics, m
         assert res == obs
 
     def test_no_eigval_error(self, mock_qubit_device_with_original_statistics):
-        """Tests that an error is thrown if expval is called with an observable that does not have eigenvalues defined."""
+        """Tests that an error is thrown if expval is called with an observable that does
+        not have eigenvalues defined."""
         dev = mock_qubit_device_with_original_statistics()
+
+        # observable with no eigenvalue representation defined
+        class MyObs(qml.operation.Observable):
+            num_wires = 1
+
+            @property
+            def eigvals(self):
+                return None
+
+        obs = MyObs(wires=0)
+
         with pytest.raises(ValueError, match="Cannot compute analytic expectations"):
-            dev.expval(qml.Hamiltonian([1.0], [qml.PauliX(0)]))
+            dev.expval(obs)
```

Comment:
Changed this because it is safer to make an op that has no evals defined (since we may change this for Hamiltonian)